scannerlib/configchecks/group_criteria.go

```diff
@@ -215,3 +223,65 @@ func getCmpStr(gc *ipb.GroupCriterion) string {
 	}
 	return strconv.Itoa(int(gc.GetConst()))
 }
+type lessThanVersionMatcher struct {
+	cmpStr string
+}
+func (m *lessThanVersionMatcher) match(group string) bool {
+	if !swVersRe.MatchString(group) {
+		log.Printf("unable to parse %q as a software version", group)
+		return false
+	}
+	version := strings.ReplaceAll(m.cmpStr, "-", ".")
+	re := regexp.MustCompile("\\W+")
+	chunks_group := re.Split(group, -1)
+	chunks_version := re.Split(version,-1)
+	min_len := 0
+	if len(chunks_group) < len(chunks_version) {
+		min_len = len(chunks_group)
+	} else {
+		min_len = len(chunks_version)
+	}
+	for i:=0; i< min_len; i++ {
+		chunks_group[i] = fmt.Sprintf("%06s", chunks_group[i])
+		chunks_version[i] = fmt.Sprintf("%06s", chunks_version[i])
+		if chunks_group[i] == chunks_version[i] {
+			continue
+		}
+		return chunks_group[i] < chunks_version[i]
+	}
+	return false
+}
+func (m *lessThanVersionMatcher) String() string {
+	return "< " + m.cmpStr
+}
+type greaterThanVersionMatcher struct {
+	cmpStr string
+}
+func (m *greaterThanVersionMatcher) match(group string) bool {
+	if !swVersRe.MatchString(group) {
+		log.Printf("unable to parse %q as a software version", group)
+		return false
+	}
+	version := strings.ReplaceAll(m.cmpStr, "-", ".")
+	re := regexp.MustCompile("\\W+")
+	chunks_group := re.Split(group, -1)
+	chunks_version := re.Split(version,-1)
+	min_len := 0
+	if len(chunks_group) < len(chunks_version) {
+		min_len = len(chunks_group)
+	} else {
+		min_len = len(chunks_version)
+	}
+	for i:=0; i< min_len; i++ {
+		chunks_group[i] = fmt.Sprintf("%06s", chunks_group[i])
+		chunks_version[i] = fmt.Sprintf("%06s", chunks_version[i])
+		if chunks_group[i] == chunks_version[i] {
+			continue
+		}
+		return chunks_group[i] > chunks_version[i]
+	}
+	return false
+}
+func (m *greaterThanVersionMatcher) String() string {
```

Comment:
Please add some unit tests to group_criteria_test.go for these new criteria - similar to the tests we have for the existing ones like LESS_THAN/GREATER_THAN